src/main/java/com/amazon/services/UserServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.amazon.services;
+
+import com.amazon.dto.UserDto;
+import com.amazon.repository.IUserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+public class UserServiceImpl implements IUserService{
+
+    @Autowired
+    IUserRepository repository;
+
+    @Override
+    public UserDto saveUser(UserDto userDto) {
+        userDto.setId(UUID.randomUUID());
+        this.repository.save(userDto.toEntity());
```

Comment:
before saving make sure first name and password are not empty

Reply:
done